src/components/core/ContextSelector/index.tsx

```diff
@@ -1,177 +1,139 @@
+                    onClick={() => !isOpen && setIsOpen(true)}
+                    placeholder={selectedValue !== '' ? selectedValue : 'Search contexts'}
+                    className={styles.searchInput}
+                    ref={inputRef}
+                />
+                {isQuerying && <Spinner inline />}
+            </>
+        );
     });
 
-    const getButtonContent = () => {
-        if (currentContextForType) {
-            return currentContextForType.title;
-        }
-
-        if (isFetchingRelatedContexts) {
-            return (
-                <span>
-                    Resolving {currentContext ? currentContext.title : null} <Spinner inline />
-                </span>
-            );
-        }
+    const updateOptions = (item: SearchableDropdownSection) =>
+        dropdownSections.map(option => {
+            return { ...option, isSelected: item.key === option.key };
+        });
 
-        if (currentContext && !isFetchingRelatedContexts) {
-            return `Unable to resolve ${currentContext.title}. Please select context`;
-        }
+    const { isOpen, setIsOpen, controllerRef } = dropdownController;
 
-        return 'Select context';
-    };
+    const onSelect = React.useCallback(
+        item => {
+            if (item.key && item.key === 'empty') {
+                return;
+            }
+            if (isOpen) {
+                const selectedContext = contexts.find(c => c.id === item.key);
+                updateOptions(item);
+                setIsOpen(false);
+                setQueryText('');
+                setCurrentContextAsync(selectedContext);
+            }
+        },
+        [isOpen, contexts]
+    );
 
     const containerClassNames = classNames(styles.container, useComponentDisplayClassNames(styles));
+    const containerRef = controllerRef as React.MutableRefObject<HTMLDivElement | null>;
+    const helperText = React.useMemo(
+        () =>
+            !contexts.length && !isQuerying && !queryText
+                ? 'No contexts, please try the search bar above (Start typing to search)'
+                : null,
+        [contexts, isQuerying, queryText]
+    );
 
     return (
-        <div ref={containerRef} onClick={onOpen} className={containerClassNames}>
-            <SearchIcon color="#DADADA" />
-            {shouldShowResultsDropdown ? (
-                <>
-                    <input
-                        ref={setInputRef}
-                        type="text"
-                        value={queryText}
-                        onChange={onSearchChange}
-                        onKeyUp={onKeyUp}
-                        placeholder={'Search'}
-                        className={inputClassNames}
-                    />
-                    {isQuerying && <Spinner inline />}
-                </>
-            ) : (
-                <button className={styles.contextButton}>{getButtonContent()}</button>
-            )}
-        </div>
+        <HeaderContentPortal>

```

Comment:
The context selector should _not_ controll where it's rendered. Thats up to the user of the component